model_rnn/dataset.py

```diff
@@ -0,0 +1,37 @@
+import pandas as pd
+import torch
+import torch.nn as nn
+import torch.utils.data as data
+from .vectorize import make_input_vect
+from .preprocessor import normalize_string
+from .config import DATA_FILEPATHS
+from .config import DATA_CHAR_TO_IX
+
+
+class Dataset(data.Dataset):
+    def __init__(self):
+        self.titles = [normalize_string(x) for x in self.read_data()]
+        self.preprocess()
+    
+    def __len__(self):
+        return len(self.X)
+    
+    def __getitem__(self, index):
+        return self.X[index], self.y[index]
+
+    def read_data(self) -> list[str]:
+        return pd.read_csv(DATA_FILEPATHS['item_master'])['PRODUCT_NAME']\
+            .dropna().astype(str).tolist()
```

Comment:
Also can be in init, use `Series.apply(preprocess_text)` while it's still a pandas series.